### PR TITLE
Scala parser update

### PR DESCRIPTION
**Pull request by njohnwalker**
Updated the scala parser to accept meta sort declarations and the `\next` pattern production

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance

---

### CODE REVIEW

File: src/main/scala/org/kframework/kore/Interface.scala [hunk 1]
```diff
@@ -35,15 +35,15 @@ object Import {
 }
 
 trait SortDeclaration extends Declaration {
-  def params: Seq[SortVariable]
+  def params: Seq[Sort]
```
Comment (virgil-serbanuta) [inline src/main/scala/org/kframework/kore/Interface.scala]: Why this change? I think that the semantics-of-k doc specifies a variable here.

**Comment by njohnwalker** [comment]
As discussed, I am closing this pull request because we wish to freeze development of the Scala parser for the time being, with it's implementation in sync with a specific revision of the Semantics of K document. (This update would be a branch from the semantics of k document)
